pyproject.toml

```diff
@@ -0,0 +1,135 @@
+[build-system]
+requires = [
+    "setuptools >= 61.0",
+    "wheel",
+    "setuptools-scm[toml]>=6.0"
+   ]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "tubular"
+version = "1.3.1"
+dependencies = [
+    "pandas>=1.5.0",
+    "scikit-learn>=1.2.0"
+]
+requires-python = ">=3.8"
+authors = [{ name = "LV GI Data Science Team", email="#DataSciencePackages@lv.co.uk"}]
+description = "Package to perform pre processing steps for machine learning models"
+readme = {file = "README.md", content-type = "text/markdown"}
+license = {file = "LICENSE"}
+keywords = ["data science", "feature engineering", "data transforms", "pipeline", "sklearn", "machine learning", "ML", "DS"]
+classifiers=[
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Operating System :: OS Independent",
+    "License :: OSI Approved :: BSD License",
+]
+
+[project.optional-dependencies]
+dev = [
+    "test-aide>=0.1.0",
+    "pytest>=5.4.1",
+    "pytest-mock>=3.5.1",
+    "pytest-cov>=2.10.1",
+    "pre-commit==2.15.0 ",
+    "ruff==0.2.2",
+    ]
+
+[project.urls]
+Documentation = "https://tubular.readthedocs.io/en/latest/index.html"
+Repository = "https://github.com/lvgig/tubular"
+Issues = "https://github.com/lvgig/tubular/issues"
+Changelog = "https://github.com/lvgig/tubular/CHANGELOG.md"
+
+[tool.ruff]
+# Same as Black.
+line-length = 88
+
+# Assume Python 3.8.
+target-version = "py38"
```

Comment:
S: Worth us targeting py39 explicitly now that 3.8 goes EoL in October of this year? Or should we kick that into another ticket?

Reply:
Right, I've orientated myself with the PR a bit and now understand that:

- We run CI against pythons 3.8 through 3.12 and would like to continue to support all of these versions as we have the potential for users on all pys.
- [scipy 1.10.1](https://pypi.org/project/scipy/1.10.1/) (released in Feb'23) supports 3.8<=python<3.12 and [scipy 1.11.1](https://pypi.org/project/scipy/1.11.1/) (released Jun'23) supports 3.9<=python<3.13. There is therefore no way to cover all of the python versions with a single pinned `requirements.txt`.
  - People do have debates about using multiple `requirements.txt` files (see jazzband/pip-tools#1326), but I don't really think that we should get into that.
- We still want to maintain a pinned set of requirements for users to develop against.

My proposal would be that we accept this PR's suggestion and install against `pyproject.toml` for now. We should advertise that our next release will be the last to support python 3.8 ahead of its EoL status inbound this October. At that stage, we shall drop py38 support, likely add a 3.13 testing stage and then switch to using pinned requirements again (we'll probably need to use [scipy-1.11.4](https://pypi.org/project/scipy/1.11.4/), which requires py39+).

Let's write as a spin-out ticket and then progress with this PR.

Reply:
Right, I've orientated myself with the PR a bit and now understand that:

- We run CI against pythons 3.8 through 3.12 and would like to continue to support all of these versions as we have the potential for users on all pys.
- [scipy 1.10.1](https://pypi.org/project/scipy/1.10.1/) (released in Feb'23) supports 3.8<=python<3.12 and [scipy 1.11.1](https://pypi.org/project/scipy/1.11.1/) (released Jun'23) supports 3.9<=python<3.13. There is therefore no way to cover all of the python versions with a single pinned `requirements.txt`.
  - People do have debates about using multiple `requirements.txt` files (see jazzband/pip-tools#1326), but I don't really think that we should get into that.
- We still want to maintain a pinned set of requirements for users to develop against.

My proposal would be that we accept this PR's suggestion and install against `pyproject.toml` for now. We should advertise that our next release will be the last to support python 3.8 ahead of its EoL status inbound this October. At that stage, we shall drop py38 support, likely add a 3.13 testing stage and then switch to using pinned requirements again (we'll probably need to use [scipy-1.11.4](https://pypi.org/project/scipy/1.11.4/), which requires py39+).

Let's write as a spin-out ticket and then progress with this PR.

Reply:
Right, I've orientated myself with the PR a bit and now understand that:

- We run CI against pythons 3.8 through 3.12 and would like to continue to support all of these versions as we have the potential for users on all pys.
- [scipy 1.10.1](https://pypi.org/project/scipy/1.10.1/) (released in Feb'23) supports 3.8<=python<3.12 and [scipy 1.11.1](https://pypi.org/project/scipy/1.11.1/) (released Jun'23) supports 3.9<=python<3.13. There is therefore no way to cover all of the python versions with a single pinned `requirements.txt`.
  - People do have debates about using multiple `requirements.txt` files (see jazzband/pip-tools#1326), but I don't really think that we should get into that.
- We still want to maintain a pinned set of requirements for users to develop against.

My proposal would be that we accept this PR's suggestion and install against `pyproject.toml` for now. We should advertise that our next release will be the last to support python 3.8 ahead of its EoL status inbound this October. At that stage, we shall drop py38 support, likely add a 3.13 testing stage and then switch to using pinned requirements again (we'll probably need to use [scipy-1.11.4](https://pypi.org/project/scipy/1.11.4/), which requires py39+).

Let's write as a spin-out ticket and then progress with this PR.

Reply:
Captured as #282. Nothing needed for this PR here (unless there's a point of disagreement with above).

Reply:
I agree. Best place to advertise?  tried adding a warning to __init__.py but doesn't seem to raise anything when doing an editable install in python 3.8. Tried both:

`if sys.version_info < (3,9):
    raise RuntimeError ("Support for python versions < 3.9 will be deprecated soon")`

`if sys.version_info < (3,9):
    warn ("Support for python versions < 3.9 will be deprecated soon")`
    
Maybe enough to just put it in the changelog/release notes?